src/utils.ts

```diff
@@ -62,13 +62,16 @@ export function get(object: object, path: PathParam) {
   return result;
 }
 
-// TODO: improve types
-export function set(object: Record<any, any>, path: PathParam, value: unknown) {
+export function set(
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  object: Record<string, any>,
+  path: PathParam,
+  value: unknown
+) {
   if (typeof path === 'string') path = stringToPath(path);
 
-  // TODO: improve types
   function createNewElement(
-    object: Record<any, any>,
+    object: Record<string, unknown>,
```

Comment:
Why do you use `any` at the set function and `unknown` at the createNewElement function?

```ts
let vAny: any = 10;          // We can assign anthing to any
let vUnknown: unknown =  10; // We can assign anthing to unknown just like any 


let s1: string = vAny;     // Any is assigable to anything 
let s2: string = vUnknown; // Invalid we can't assign vUnknown to any other type (without an explicit assertion)
```

Reply:
At first i prefer to use `unknown`, but it may be a headache on some expressions like `object[acessor] = newValue`
so, to avoid some rage moments with typescript i use `any`

Reply:
can i show you the problems later?

Reply:
Sure, I approved the PR. We can talk about this later.
I'm Just waiting for Vinicius's approval to make the merge.

Reply:
I have to work on some requested changes.
After approval i can text youi :)